mirascope/gemini/tools.py

```diff
@@ -68,20 +73,42 @@ def tool_schema(cls) -> Tool:
         Returns:
             The constructed `Tool` schema.
         """
-        tool_schema = super().tool_schema()
-        if "parameters" in tool_schema:
-            if "$defs" in tool_schema["parameters"]:
-                raise ValueError(
-                    "Unfortunately Google's Gemini API cannot handle nested structures "
-                    "with $defs."
+        super().tool_schema()
+        model_schema: dict[str, Any] = cls.model_json_schema()
+        pprint.pprint(model_schema)
+
+        # Replace all references with their values
+        without_refs: dict[str, Any] = jsonref.replace_refs(model_schema)  # type: ignore
+        pprint.pprint(without_refs)
+
+        # Remove all Defs
+        without_refs.pop("$defs")
+        pprint.pprint(without_refs)
+
+        # Get the name and description, and remove them from the schema
+        name: str = without_refs.pop("title")  # type: ignore
```

Comment:
quick note:

name and description are already in `super().tool_schema()`, so it's really just the `parameters` field that needs to be modified (and thus remove duplicate code).